// @superset-ui/core/esm/dashboard/types/Base.js
(function () {var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;enterModule && enterModule(module);})();var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default.signature : function (a) {return a;};








































export let NativeFilterType;(function (NativeFilterType) {NativeFilterType["NATIVE_FILTER"] = "NATIVE_FILTER";NativeFilterType["DIVIDER"] = "DIVIDER";})(NativeFilterType || (NativeFilterType = {}));




export let DataMaskType;(function (DataMaskType) {DataMaskType["NativeFilters"] = "nativeFilters";DataMaskType["CrossFilters"] = "crossFilters";})(DataMaskType || (DataMaskType = {}));






















































export function isNativeFilter(
filterElement)
{
  return filterElement.type === NativeFilterType.NATIVE_FILTER;
}

export function isFilterDivider(
filterElement)
{
  return filterElement.type === NativeFilterType.DIVIDER;
};(function () {var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;if (!reactHotLoader) {return;}reactHotLoader.register(isNativeFilter, "isNativeFilter", "/Users/ville/src/superset-release/superset-frontend/packages/superset-ui-core/src/dashboard/types/Base.ts");reactHotLoader.register(isFilterDivider, "isFilterDivider", "/Users/ville/src/superset-release/superset-frontend/packages/superset-ui-core/src/dashboard/types/Base.ts");})();;(function () {var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;leaveModule && leaveModule(module);})();
