// @superset-ui/core/lib/dashboard/types/Base.js
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
export var NativeFilterType;
(function (NativeFilterType) {
    NativeFilterType["NATIVE_FILTER"] = "NATIVE_FILTER";
    NativeFilterType["DIVIDER"] = "DIVIDER";
})(NativeFilterType || (NativeFilterType = {}));
export var DataMaskType;
(function (DataMaskType) {
    DataMaskType["NativeFilters"] = "nativeFilters";
    DataMaskType["CrossFilters"] = "crossFilters";
})(DataMaskType || (DataMaskType = {}));
export function isNativeFilter(filterElement) {
    return filterElement.type === NativeFilterType.NATIVE_FILTER;
}
export function isFilterDivider(filterElement) {
    return filterElement.type === NativeFilterType.DIVIDER;
}
