// @superset-ui/core/esm/chart/types/Base.js
(function () {var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;enterModule && enterModule(module);})();var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default.signature : function (a) {return a;};























export let Behavior;(function (Behavior) {Behavior["INTERACTIVE_CHART"] = "INTERACTIVE_CHART";Behavior["NATIVE_FILTER"] = "NATIVE_FILTER";})(Behavior || (Behavior = {}));




export let AppSection;(function (AppSection) {AppSection["EXPLORE"] = "EXPLORE";AppSection["DASHBOARD"] = "DASHBOARD";AppSection["FILTER_BAR"] = "FILTER_BAR";AppSection["FILTER_CONFIG_MODAL"] = "FILTER_CONFIG_MODAL";AppSection["EMBEDDED"] = "EMBEDDED";})(AppSection || (AppSection = {}));
























export let ChartLabel;(function (ChartLabel) {ChartLabel["DEPRECATED"] = "DEPRECATED";ChartLabel["FEATURED"] = "FEATURED";})(ChartLabel || (ChartLabel = {}));




export const chartLabelExplanations = {
  [ChartLabel.DEPRECATED]:
  'This chart uses features or modules which are no longer actively maintained. It will eventually be replaced or removed.',
  [ChartLabel.FEATURED]:
  'This chart was tested and verified, so the overall experience should be stable.' };


export const chartLabelWeight = {
  [ChartLabel.DEPRECATED]: {
    weight: -0.1 },

  [ChartLabel.FEATURED]: {
    weight: 0.1 } };const _default =



{};export default _default;;(function () {var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;if (!reactHotLoader) {return;}reactHotLoader.register(chartLabelExplanations, "chartLabelExplanations", "/Users/ville/src/superset-release/superset-frontend/packages/superset-ui-core/src/chart/types/Base.ts");reactHotLoader.register(chartLabelWeight, "chartLabelWeight", "/Users/ville/src/superset-release/superset-frontend/packages/superset-ui-core/src/chart/types/Base.ts");reactHotLoader.register(_default, "default", "/Users/ville/src/superset-release/superset-frontend/packages/superset-ui-core/src/chart/types/Base.ts");})();;(function () {var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;leaveModule && leaveModule(module);})();
